website/blog/2021-04-26-7.14.0.md

```diff
@@ -0,0 +1,165 @@
+exports.default = two;
+
+function two() {
+  return 2;
+}
+```
+
+When importing this library with Babel (or TypeScript, Rollup or similar tools) and compiling your code to CommonJS, it will look like:
+
+```js
+import two from "two";
+console.log(two());
+```
+
+One day, you decide to provide two versions of your code: a compiled CommonJS one, and one using native ECMAScript modules.
+
+While the compiled version works, the ESM one will throw `TypeError: two is not a function`. This is because in Node.js, the default import is _not_ the dependency's `exports.default`, but the whole `module.exports.default` object instead.
+
+You could change your code to:
+
+```js
+import two from "two";
+console.log(two.default());
+```
+
+However, this new code has a problem: it now doesn't work when compiled, because `two.default` is not a function.
+
+Babel v7.14.0 adds a new `importInterop: "node"` option in the [`@babel/plugin-transform-commonjs`](https://babeljs.io/docs/en/babel-plugin-transform-commonjs) plugin that allows `import` statements to match the native Node.js behavior. You can read more about this option [in the docs](https://babeljs.io/docs/en/babel-plugin-transform-modules-commonjs#importInterop).
+
+[Nicolò](https://github.com/nicolo-ribaudo) from our team also [contributed](https://github.com/rollup/plugins/pull/838) a similar option to `@rollup/plugin-commonjs`, which will be out in the next release. Our goal is to help the ecosystem migrate to native ECMAScript modules by providing an easier migration path.
+
+### TypeScript 4.3
+
+The new TypeScript version, which will be released as stable in May, supports a few new features:
+
+- `override` modifiers in class elements
+- static index signatures (`[key: KeyType]: ValueType`) in classes
+- `get`/`set` in type declarations
+
+You can read more about them in the [TypeScript 4.3 release post](https://devblogs.microsoft.com/typescript/announcing-typescript-4-3/). This is supported through [@babel/preset-typescript](https://babeljs.io/docs/en/babel-preset-typescript).
+
+### `async do` expressions
+
+[`async do` expressions](https://github.com/tc39/proposal-async-do-expressions) are a Stage 1 proposal built on top of the [`do` expressions](https://github.com/tc39/proposal-do-expressions) proposal.
+
+They allow using asynchronous blocks within synchronous code, and those blocks are evaluated as a promise:
+
+```js
+function sync() {
+  let x = async do {
+    let res = await Promise.resolve("Third!");
+    console.log("Second!");
+    res;
+  };
+  console.log("First!");
+  x.then(console.log);
+}
+
+console.log(sync());
+// Logs:
+// - "First!"
+// - "Second!"
+// - "Third!"
+```
+
+You can test this proposal (and report feedback!) by adding the [`@babel/plugin-proposal-do-expressions`](https://babeljs.io/docs/en/babel-plugin-proposal-do-expressions) and `@babel/plugin-proposal-async-do-expressions` plugins to your Babel configuration.
```

Comment:
Was going to link this if we make a doc for it `@babel/plugin-proposal-async-do-expressions`

Reply:
https://github.com/babel/website/pull/2486

```suggestion
You can test this proposal (and report feedback!) by adding the [`@babel/plugin-proposal-do-expressions`](https://babeljs.io/docs/en/babel-plugin-proposal-do-expressions) and [`@babel/plugin-proposal-async-do-expressions`](https://babeljs.io/docs/en/babel-plugin-proposal-async-do-expressions) plugins to your Babel configuration.
```